Jenkinsfile

```diff
@@ -48,129 +44,103 @@ pipeline {
-                                withMaven(maven: 'Maven 3.3.9', globalMavenSettingsConfig: 'default-global-settings', mavenSettingsConfig: 'codice-maven-settings', mavenOpts: '${LARGE_MVN_OPTS} ${LINUX_MVN_RANDOM}') {
-                                    sh 'mvn clean install -B -T 1C -pl !$ITESTS'
-                                    sh 'mvn install -B -pl $ITESTS -nsu'
-                                }
+            parallel {
+                stage ('Linux') {
+                    steps {
+                        timeout(time: 3, unit: 'HOURS') {
+                            // TODO: Maven downgraded to work around a linux build issue. Falling back to system java to work around a linux build issue. re-investigate upgrading later
+                            withMaven(maven: 'Maven 3.3.9', globalMavenSettingsConfig: 'default-global-settings', mavenSettingsConfig: 'codice-maven-settings', mavenOpts: '${LARGE_MVN_OPTS} ${LINUX_MVN_RANDOM}') {
+                                sh 'mvn clean install -B -T 1C -pl !$ITESTS'
+                                sh 'mvn install -B -pl $ITESTS -nsu'
                             }
                         }
-                    },
-                    windows: {
-                        node('proxmox-windows') {
-                            bat 'git config --system core.longpaths true'
-                            retry(3) {
-                                checkout scm
-                            }
-                            timeout(time: 3, unit: 'HOURS') {
-                                withMaven(maven: 'M35', jdk: 'jdk8-latest', globalMavenSettingsConfig: 'default-global-settings', mavenSettingsConfig: 'codice-maven-settings', mavenOpts: '${LARGE_MVN_OPTS}') {
-                                    bat 'mvn clean install -B -T 1C -pl !%ITESTS%'
-                                    bat 'mvn install -B -pl %ITESTS% -nsu'
-                                }
+                    }
+                }
+                stage ('Windows') {
+                    agent { label 'server-2016-large'}
+                    steps {
+                        timeout(time: 3, unit: 'HOURS') {
+                            withMaven(maven: 'M35', jdk: 'jdk8-latest', globalMavenSettingsConfig: 'default-global-settings', mavenSettingsConfig: 'codice-maven-settings', mavenOpts: '${LARGE_MVN_OPTS}') {
+                                bat 'mvn clean install -B -T 1C -pl !%ITESTS%'
+                                bat 'mvn install -B -pl %ITESTS% -nsu'
                             }
                         }
                     }
-                )
+                }
             }
         }
         stage('Security Analysis') {
-            steps {
-                parallel(
-                    owasp: {
-                        node('linux-large') {
-                            retry(3) {
```

Comment:
@mcalcote are we losing the retry functionality by switching away from the scripted syntax? Is there a way we can advise this still?

Reply:
Adding the retry logic back into the top level agent as well as each stage where we use an alternate agent.